### PR TITLE
Delete copy=False argument in function_types.py to make compatible with numpy 2.x

### PR DESCRIPTION
**Pull request by simon-poon**

### CODE REVIEW

File: thermocouples_reference/function_types.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def __call__(self,T,derivative=0,out_of_range="raise"):
         if out_of_range not in ["raise", "nan", "extrapolate"]:
             raise ValueError("invalid out_of_range parameter",out_of_range)
 
-        T = np.array(T, copy=False, order='A')
+        T = np.array(T, order='A')
```
Comment (jayich) [inline thermocouples_reference/function_types.py:103]: Why does this change fix the issue?

Comment (simon-poon) [reply thermocouples_reference/function_types.py:103]: They changed the meaning of False, as described [here](https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword). It now throws an exception if a copy is needed, which is the case here.

**Review by jayich — APPROVED** [review]
lgtm.

**Comment by jayich** [comment]
Actually, what happens to the unused copy arguments in the function?

**Comment by simon-poon** [comment]
It defaults to True. Which I don't think matters since T is defined from T, and I also tested to check that it doesn't affect anything that we use.

**Comment by simon-poon** [comment]
@jayich pinging you because I'm not sure if you saw my answer. Is this good to merge?

**Comment by jayich** [comment]
> It defaults to True. Which I don't think matters since T is defined from T, and I also tested to check that it doesn't affect anything that we use.

I don't understand what "since T is defined from T" means...

**Comment by simon-poon** [comment]
I meant that the numpy array T is defined from another variable named T, T=np.array(T). The effect that "copy=False" has is that any changes that happened to the old variable will also happen to the new np array. But since T is recursively defined, you can't change the old variable, so there shouldn't be any unforeseen consequences of getting rid of the copy argument.

**Comment by jayich** [comment]
lgtm.

**Pull request merged**